presto-main/src/main/java/com/facebook/presto/operator/SpatialJoinOperator.java

```diff
@@ -113,14 +114,14 @@ public OperatorFactory duplicate()
     }
 
     private final OperatorContext operatorContext;
+    private final LocalMemoryContext localUserMemoryContext;
     private final List<Type> probeTypes;
     private final List<Type> outputTypes;
     private final List<Integer> probeOutputChannels;
     private final int probeGeometryChannel;
     private final PagesSpatialIndexFactory pagesSpatialIndexFactory;
 
-    private ListenableFuture<Supplier<PagesSpatialIndex>> pagesSpatialIndexFuture;
-    private PagesSpatialIndex pagesSpatialIndex;
+    private ListenableFuture<PagesSpatialIndex> pagesSpatialIndexFuture;
```

Comment:
i know you're nulling this out when finishing. Also null this out in `close()`, in case this operator is closed without finishing.